### ISSUE TITLE
German Translation

### ISSUE THREAD

**Issue by Baracudus** [diasurgical/devilutionX#905]
I could translate the whole game to native german if desired. I searched the forum for a existing translation file and found this italian "dexolutionx-it.po" file (attached) in this forum as a base to start with. Is there a "official" way to translate the whole project or is the german language done by this time?

Does the game support german characters like "ß" and "ö", "ä", "ü"?

Would like to contribute, please let me know how to do it "the right way". I there a "guide"  or something I missed? Thanks!
[devilutionx.zip](https://github.com/diasurgical/devilutionX/files/5462036/devilutionx.zip)

**Comment by AJenbo** [comment]
Hi, translation support is still in the early stage. I believe I have the basics working but still need to tweak the build scripts and mark all the strings for translation. This effort can be tracked here: https://github.com/diasurgical/devilutionX/pull/902

ß, ö, ä, and ü are part of the Latin-1 Supplement block in Unicode, which we aim to support in the first version that will include translations. Possibly utilizing the font found here as a basis:
https://github.com/diasurgical/devilution/issues/32#issuecomment-402322326

The PO file should be UTF8 encoded. Since I haven't marked all the text in the application yet, the Itialien translation would be the best starting position at this point.

There won't be a guide before things are in a more coherent state. Preferably one where you can check your work ingame. For an overall status of this you can follow this issue: https://github.com/diasurgical/devilutionX/issues/66

**Comment by ChaosMarc** [comment]
Keep in mind that there is a multilang (german included) PSX Version of Diablo. There are german patches floating around (https://www.gog.com/forum/general_de/sammelthread_zu_deutschmods_und_sprachpatches_fur_spiele_die_hier_nur_auf_englisch_erhaltlich_sin/post2038). With a little search, or the right tooling you might find the extracted strings or an iso to extract them yourself. this way you don't need to translate everything by hand

**Comment by Baracudus** [comment]
> Keep in mind that there is a multilang (german included) PSX Version of Diablo. 

Yea, I know this Version! I will compare strings etc before posting anything here when I'm not 100% sure what the exact meaning in english was for the artist (this is special to the extended conversations in Tritram) Also, some things make simply no sense in german. Like the english "of". (Sometimes). In german, this can be translated as "von", "vom", "der", "die", "das", "welches", "welchem",...etc... 

Exaple: 
"King's sword of haste" --> König's Schwert >>**der**<< Eile
"Book of Teleport" --> "Buch >>**des**<< Teleport**s**

So, I fight some time with the pure sense of many things because german is a very context-related "betwen the phrases" -like speech when it shoud sound right to natives at the end. But I'm almost half way through.

Is there a possibility to load the file into devolutionx for testing? (I read the thread from may when the translation system was not ready yet, but maybe there is a dev-update since then going on?)

**Comment by AJenbo** [comment]
Still struggling to get it working on Windows and macOS, the PSX format is also in a special format so you wouldn't be able to just load it with DevilutionX.

**Comment by gondur** [comment]
> Example:
> "King's sword of haste" --> König's Schwert >>**der**<< Eile
> "Book of Teleport" --> "Buch >>**des**<< Teleport**s**
> 


well,  if we go for ultra correct translation tries ;) 

 "King's sword of haste" --> "**Königliches** Schwert der Eile" 
"Book of Teleport" --> "Buch des **Teleportierens**" oder "Buch der **Teleportiererei**" (little bit archaic but might fit the Diablo medieval setting )

**Comment by Baracudus** [comment]
"King's sword of haste" --> "Königliches Schwert der Eile" 

Yea, that's finest german, but what to do if you translate "kings" with "königliches", when you got ... let's say:

**Comment by Baracudus** [comment]
Oh, I clicked wrong.. sorry :)

**Comment by gondur** [comment]
yes, I would help - but also my time is limited. I'm quite good and fast in reviewing - so some minor part in "rough" translation and/or bigger part / complete set on the "fine tuning" work (in fact I think, to keep it consistent someone should synchronize it then completely )

**Comment by AJenbo** [comment]
You can now translate and test the result in game. Best way to get started is to install Poedit and open https://github.com/diasurgical/devilutionX/blob/master/Translations/devilutionx.pot and then create a de.po file from it.

**Comment by Baracudus** [comment]
> You can now translate and test the result in game. Best way to get started is to install Poedit and open https://github.com/diasurgical/devilutionX/blob/master/Translations/devilutionx.pot and then create a de.po file from it.

Sounds good! When I want to test the. po file, how can I do it? Do I need a debugbuild for it or is there something like a special folder structure?

**Comment by AJenbo** [comment]
When hitting save in Poedit it will generate a .mo file, copy this to the same folder as the game.
In your ini file simply set the language code to `de` and then the text should appear ingame.

You can get the latest beta with translation support here: https://34628-143324737-gh.circle-artifacts.com/0/devilutionx_x64.zip

**Comment by popy2k14** [comment]
Any chance of getting an german translation for switch?

thx

**Comment by Baracudus** [comment]
I think when the translation is complete, it will be useable for all plattforms as it uses all the same source as I understand. Translation is going on, but Im in hospital right now, so I will continue as I retrun.

@AJenbo I managed to get the menu working in german and all Dialogs while not ingame. When I make a new game, no translation is seen there (As scroll, portion, book, witch..) curiosely, farnham the drunk is correct didplayed as Farnham der Trunkenbolt 😉 will add screenshots when I return home.

**Comment by popy2k14** [comment]
thx, for the update.
Good luck in hospital and stay healthy!!!!!!!

**Comment by AJenbo** [comment]
> @AJenbo I managed to get the menu working in german and all Dialogs while not ingame. When I make a new game, no translation is seen there (As scroll, portion, book, witch..) curiosely, farnham the drunk is correct displayed as Farnham der Trunkenbolt 😉 will add screenshots when I return home.

sounds odd make sure you use the latest beta and start a new game, you can also submit your current progress so we can look at it and others can review.

**Comment by Baracudus** [comment]
> When hitting save in Poedit it will generate a .mo file, copy this to the same folder as the game.
> In your ini file simply set the language code to `de` and then the text should appear ingame.
> 
> You can get the latest beta with translation support here: https://34628-143324737-gh.circle-artifacts.com/0/devilutionx_x64.zip

Is this a link which directs to actually newest beta automaticaly? So when I download say one time a day the exe I have always the newest beta? Ill submit when Im back home. Tristram translation is nearly complete, no translation in the inventory, not on items. All npcs and Shops work in german. Dialogs work, menu doesnt work (New game, save game...) when down in Level 1+ no translation works (Monsters, chests, books, etc)

**Comment by AJenbo** [comment]
No, it is the same version each time you use the link. You can see the latest builds here:
https://github.com/diasurgical/devilutionX/actions/workflows/Windows_x64.yml?query=branch%3Amaster+is%3Asuccess+

Currently the latest is this version. I thin everything you mentioned has been solved.

https://github.com/diasurgical/devilutionX/suites/2691019539/artifacts/59307658

Also make sure to update the translation file either using the POT template or update form source (after updating the repository).

We fixed a few strings over the past two weeks.

**Comment by Baracudus** [comment]
> No, it is the same version each time you use the link. You can see the latest builds here:
> https://github.com/diasurgical/devilutionX/actions/workflows/Windows_x64.yml?query=branch%3Amaster+is%3Asuccess+

That did the trick! Translation works for now with the latest beta release.

**Comment by aruffj** [comment]
> 
> 
> > No, it is the same version each time you use the link. You can see the latest builds here:
> > https://github.com/diasurgical/devilutionX/actions/workflows/Windows_x64.yml?query=branch%3Amaster+is%3Asuccess+
> 
> That did the trick! Translation works for now with the latest beta release.

Hey there,

Love to assist on the translation - either in proofreading or extending it. Could you share your current WIP?

**Comment by ChaosMarc** [comment]
@Baracudus Hey, I second @aruffj 's idea. Perhaps you could open a WIP Pull Request with your current version like https://github.com/diasurgical/devilutionX/pull/2135. That way others could discuss the current status in your PR and/or create PRs to your branch to help with the translation.

**Comment by ChaosMarc** [comment]
I've opened a PR for the translation. @Baracudus Please join the party and share your progress :)
https://github.com/diasurgical/devilutionX/pull/2138

**Comment by Baracudus** [comment]
Hey Marc. Sorry, I was busy the last weeks, Ill check my current status and merge the working bits with the already translated quest texts and share them soon. Be patient, content is coming 👍

**Comment by Baracudus** [comment]
So, here we go with "Version 1" of the full German transaltion (94% translated). I used the version

![version](https://user-images.githubusercontent.com/71731621/126899478-672c4637-024a-4888-b18e-467afe140331.png)

to test the strings etc.. and as my limited time allows me, I'm done with the preparings. Now, someone else is invented to take my work and "finetune" it. Engingeside problems like missing german-typical genders and pronounces cant't be adressed only by a translation file itsselfe. Something, we have to live with, I think. It's far from perfect, I tried to make the all the Menus work, the Shops, the Quests etc.. But sometimes, there are funny Items generated which I can't adress only from chenges to the translation file (like "den", "dem", "dessen", "deren", an so on..). I enjoyed it, now have fun with the files. 

Used Software:
POEdit 3.0 Free (Build 6376)

Here is the ZIP
[de.zip](https://github.com/diasurgical/devilutionX/files/6873898/de.zip)

And finally, here some screens.
![sc1_menu](https://user-images.githubusercontent.com/71731621/126899775-46f51d21-b11d-4425-bc7b-0dd32e305c8c.png)
![sc2_menusp](https://user-images.githubusercontent.com/71731621/126899778-99bfa1ce-ee05-4f75-98d3-36cb3da6b6da.png)
![screen00](https://user-images.githubusercontent.com/71731621/126899783-9a16cd4e-5728-4afb-9bf5-428474fdf92a.png)
![screen01](https://user-images.githubusercontent.com/71731621/126899784-64ef627b-fc95-496e-ab5b-d9abf2fe9fa4.png)
![screen03](https://user-images.githubusercontent.com/71731621/126899785-6648c087-ecc7-4c7a-8951-9de99aec98e2.png)
![screen04](https://user-images.githubusercontent.com/71731621/126899786-3e36f6ea-9f24-4d50-a2b4-0b520c02901e.png)
![screen05](https://user-images.githubusercontent.com/71731621/126899789-838ed465-7e5c-463a-87a6-d82395ce413a.png)
![screen06](https://user-images.githubusercontent.com/71731621/126899790-6961b3e7-3cbc-4c80-8dd9-51de0418883e.png)
![screen07](https://user-images.githubusercontent.com/71731621/126899792-22ae1dd6-096d-4e75-a19d-691afa3601e7.png)
![screen08](https://user-images.githubusercontent.com/71731621/126899793-2c81e4a0-3ead-4db2-8f54-68871df77242.png)
![screen09](https://user-images.githubusercontent.com/71731621/126899794-239f0a61-8c10-4bb7-bc97-84cffb34897b.png)
![screen10](https://user-images.githubusercontent.com/71731621/126899796-c07d503e-2b2e-44ce-a9fa-ccaf8a60fb29.png)

**Comment by ChaosMarc** [comment]
Thank you for this huge piece of work! I've committed your version (unmodified) in my [Pull Request](https://github.com/diasurgical/devilutionX/pull/2138). All credits go to you. Is it safe to assume that due to the fact that you didn't commit it via Git yourself you don't know how. For further collaboration I strongly recommend that you learn it, as it will make everyone's life much easier. If you need help with that just ask in Discord (you can send me direct message if you want the help to be in German 👼)

**Comment by Baracudus** [comment]
@ChaosMarc Thx 👍 I'll try to learn this whole devstuff things (sure, I'm a nerd since around 1994) but so much has changed and is new to me when it comes to development. I'm a classic C++/C# guy and NX-Constructor theese days.. But I love this game since release day (Got my CD at "day one" in germany,.) so, making some dev/improvements to it is very welcome for me. 

Hope it will work, there are some things I would change, and sometimes it would be nice if the font would scale a "little" bit down for cut off things like "Haltbar:" --> "Haltbarkeit:" because this just doesn't fit in the window without dropping out of the frame (which looks odd). Also, (maybe that works now with newer betas) the chars öÖäÄüÜ have to work to make real "german" sense.

**Comment by AJenbo** [comment]
öÖäÄüÜß etc will be supported by the time of the next release so please use them in your translation :)

if you could upload the translation here then the lines will be correctly attributed to you rather then ChaosMarc https://github.com/diasurgical/devilutionX/upload/master/Translations

**Comment by Baracudus** [comment]
Whem I click the link, it says..

_Uploads are disabled.
File uploads require **push access** to this repository._

.. what to do? 

> please use them in your translation

I used "normal" German on the whole thing -> done

**Comment by AJenbo** [comment]
> _Uploads are disabled. File uploads require **push access** to this repository._
> 
> .. what to do?

Hmm unfortunately it seams that uploading a file isn't as straight forward as editing on :(

To do it you will first need to hit "Fork" on the project page:
![image](https://user-images.githubusercontent.com/204594/126913801-1bfa9ef4-43d4-47e0-97a7-bef0a11fdd4c.png)
(https://github.com/diasurgical/devilutionX top right)

You should then be able to upload the file here:
https://github.com/Baracudus/devilutionX/upload/master/Translations

You can then open a Pull Request:
https://github.com/diasurgical/devilutionX/compare/master...Baracudus:master

**Comment by AJenbo** [comment]
I have gone ahead and merged the translation via the PR made by @ChaosMarc thanks for the work on this :)

**Comment by AJenbo** [comment]
P.s. it would also be awesome if you could translate the strings for the Android port, there aren't that many but I pertains to how to install the game data so it would probably be really good to have in peoples locale language.

**Comment by AJenbo** [comment]
Seams the translation was done on an older template, I have update the translation file to match the latest content, about 21% of the strings still needs to be translated.

**Comment by Baracudus** [comment]
I'll translate the files for Android and the updated one and repost them here 😎👍

**Comment by AJenbo** [comment]
German letter support is almost here:

![image](https://user-images.githubusercontent.com/204594/128099900-a0301639-0ff7-4881-b179-d9b98b43a925.png)

**Comment by AJenbo** [comment]
Attacked is the community Hellfire translation provided by sheng_luwei. It's a bit unstructured since it's stripped from a hex translation but hoping it might still be helpful for finishing the German translation :)

[hellfire_de.txt](https://github.com/diasurgical/devilutionX/files/7071223/hellfire_de.txt)

**Comment by Baracudus** [comment]
Translation of hellfire_de.txt, de.po(new) and strings.txt -> next week 😊👍
